File: on-boarding-samples/on-boarding-sample-java/src/main/java/apio/architect/example/Person.java
package apio.architect.example;

import com.liferay.apio.architect.annotation.Id;
import com.liferay.apio.architect.annotation.Vocabulary.Field;
import com.liferay.apio.architect.annotation.Vocabulary.Type;
import com.liferay.apio.architect.identifier.Identifier;

@Type("Person")
public interface Person extends Identifier<Long> {

    @Field("name")
    String getName();

    @Field("jobTitle")
    String getJobTitle();

    @Id
    long getId();

    static Person of(long id, String name, String jobTitle) {
        return new Person() {
            @Override
            public String getName() {
                return name;
            }

            @Override
            public String getJobTitle() {
                return jobTitle;
            }

            @Override
            public long getId() {
                return id;
            }
        };
    }

}

File: on-boarding-samples/on-boarding-sample-kotlin/src/main/java/apio/architect/example/Person.kt
package apio.architect.example

import com.liferay.apio.architect.annotation.Id
import com.liferay.apio.architect.annotation.Vocabulary
import com.liferay.apio.architect.annotation.Vocabulary.Field
import com.liferay.apio.architect.identifier.Identifier

@Vocabulary.Type("Person")
interface Person : Identifier<Long> {

    @get:Field("name") val name: String

    @get:Field("jobTitle") val jobTitle: String

    @get:Id val id: Long

    companion object {
        fun of(id: Long, name: String, jobTitle: String) = object : Person {
            override val name: String = name

            override val jobTitle: String = jobTitle

            override val id: Long = id
        }
    }
}